api/src/main/java/org/openmrs/module/registrationcore/api/mpi/openempi/PatientIdentifierMapper.java

```diff
@@ -13,32 +14,50 @@
 public class PatientIdentifierMapper {
 
     private final Log log = LogFactory.getLog(this.getClass());
-    private static final String SPLITTER_SIGN = ":";
-    //first element ("key") - local, second element ("value") - mpi.
-    private List<IdentifierMapPair> MAPPED_ID;
+    //first element - local, second element - mpi, third element - type
+    private List<IdentifierTypeMap> MAPPED_ID;
 
     @Autowired
     @Qualifier("registrationcore.mpiProperties")
     private MpiProperties mpiProperties;
 
-    public Integer getMappedLocalIdentifierTypeId(Integer mpiIdentifierTypeId) {
+    public String getMappedLocalIdentifierTypeUuid(String mpiIdentifierTypeId) {
         validateInit();
-        for (IdentifierMapPair pair : MAPPED_ID) {
-            if (pair.mpiIdentifierId.equals(mpiIdentifierTypeId))
-                return pair.localIdentifierId;
+        for (IdentifierTypeMap typeMap : MAPPED_ID) {
+            if (typeMap.mpiIdentifierId.equals(mpiIdentifierTypeId)) {
```

Comment:
I think you be avoiding using identifier type id and in general ids, use uuids

Reply:
agreed, actually in this case its referring to the MPI identifier uuid anyways, I have refactored the variable name to mpiIdentifierUuid